lib/integrations/llm_instrumentation.rb

```diff
@@ -2,9 +2,19 @@
 
 require 'opentelemetry_config'
 require_relative 'llm_instrumentation_constants'
+require_relative 'llm_instrumentation_helpers'
 
 module Integrations::LlmInstrumentation
   include Integrations::LlmInstrumentationConstants
+  include Integrations::LlmInstrumentationHelpers
+
+  PROVIDER_PREFIXES = {
+    'openai' => %w[gpt- o1 o3 o4 text-embedding- whisper- tts-],
+    'anthropic' => %w[claude-],
+    'google' => %w[gemini-],
+    'mistral' => %w[mistral- codestral-],
+    'deepseek' => %w[deepseek-]
+  }.freeze
```

Comment:
Just be careful about one thing, RubyLLM has model name verification, so you need to periodically refresh it's model list. We can disable the behaviour, but it comes with some caveats. 

Read more: https://rubyllm.com/models/